src/pdfkernel.cpp

```diff
@@ -2828,3 +2828,17 @@ wxPdfDocument::SetFillGradient(double x, double y, double w, double h, int gradi
                wxString(_("Gradient Id out of range.")));
   }
 }
+
+wxString
+wxPdfDocument::MakeFontKey(const wxString& fontName, const wxString& fontEncoding)
+{
+  wxString fontKey;
+  fontKey.reserve(fontName.Length() + fontEncoding.length() + 2);
+
+  fontKey.Append(fontEncoding.Lower());
```

Comment:
This is, of course, wrong and `fontName` should be used here. I didn't notice it, luckily someone else did in the upstream, but please do test all the changes, even the "trivial" ones like `MakeFontKey()` refactoring.

P.S. I'll fix the bug myself, but it would have been, of course, much better to avoid introducing it in the first place.

Reply:
Well, it would have been my duty to check the PR more carefully, before accepting it.

Main thing is the bug was detected and is now fixed.

In the meantime I applied another change to wxPdfDC to support drawing (rotated) multiline text with or without background color. I'm not sure whether caching the background color would make sense (like it was introduced by you for the text color). Maybe you could check this modification in your environment, too.

Reply:
Not that it excuses anything, but the original version of the change didn't use `MakeFontKey()` function and we tested everything with it (and it worked correctly, of course) and _then_, before submitting it, we refactored the code to use the new function and this is when the bug crept in. I really should have known better than to submit it without testing, of course...

As for rotated text, we don't use it currently, although it certainly could be useful in the future, so thanks for implementing support for it!